Samples/resource-owner-password-credential/Angular2/Angualar2-Client-ROPC/app/authorize/authorize-component.ts

```diff
@@ -0,0 +1,239 @@
+            this.loginBool = false;
+            this.registerBool = false;
+            this.modal.open();// check if refresh key is present it wont be present for external logged in users
+            this.refreshLogin(); // renew auth key and redirect
+        }
+        else {
+            this.refreshBool = false;
+            this.loginBool = true;
+            this.registerBool = false;
+            this.modal.open();
+        }
+    }
+
+    public logstatus() {
+        this.isLoggedin = true;
+    }
+
+    public callLogin() { // method to open login form
+        this.loginBool = true;
+        this.refreshBool = false;
+        this.registerBool = false;
+    }
+
+    public callRegister() {//method to open register form
+        this.loginBool = false;
+        this.refreshBool = false;
+        this.registerBool = true;
+    }
+    // end
+
+    public login(creds: LogModel) {
+        this.isLoading = true;
+        var instance = this;
+        this.authService.login(creds)
+            .subscribe(
+            TokenResult => {
+                this.logMsg = "You are logged In Now , Please Wait ....";
+                localStorage.setItem("auth_key", TokenResult.access_token);
+                localStorage.setItem("refresh_key", TokenResult.refresh_token);
+                instance.getUserFromServer();
+                this.mclose();
+                this.parentRouter.navigate(['/dashboard']);
+                this.isLoading = false;
+            },
+            Error => {
+                this.isLoading = false;
+                var error, key;
+                key = Error.json();
+                for (error in key) {
+                    this.logMsg = "";
+                    this.logMsg = this.logMsg + key[error] + "\n";
+                }
+            })
+    }
+
+    public refreshLogin() {
+        this.isLoading = true;
+        var instance = this;
+        this.authService.refreshLogin()
+            .subscribe(
+            TokenResult => {
+                this.isLoading = false;
+                this.logMsg = "You are logged In Now , Please Wait ....";
+                localStorage.setItem("auth_key", TokenResult.access_token);
+                localStorage.setItem("refresh_key", TokenResult.refresh_token);
+                instance.getUserFromServer();
+                this.payloader = "Authorized";
+                this.mclose();
+                this.parentRouter.navigate(['/dashboard']);
+            },
+            Error => {
+                this.isLoading = false;
+                var error, key;
+                key = Error.json();
+                for (error in key) {
+                    this.logMsg = "";
+                    this.logMsg = this.logMsg + key[error] + "\n";
+                }
+            })
+    }
+
+    public getUserFromServer() {
+        this.isLoading = true;
+        var instance = this;
+        this.resourceService.getUserInfo().subscribe(data => {
```

Comment:
you are using mixed case parameters when calling `subscribe` e.g. `.subscribe(
            TokenResult => {` vs `.subscribe(data` please make it consistent